src/ts_emit.test.iced

```diff
@@ -22,4 +35,300 @@ describe "TypescriptEmitter", () ->
       """)
       return
     return
+
+  describe "emit_imports", () ->
+    it "should not modify the given path", () ->
+      imports = [
+        {
+          path: "../gregor1",
+          type: "idl",
+          import_as: "gregor1"
+        },
+        {
+          path:  "github.com/keybase/client/go/protocol/keybase1",
+          type: "idl",
+          import_as: "keybase1"
+        }
+      ]
+      emitter.emit_imports {imports, messages: {}, types: []}, 'location/of/my/output.go'
+      code = emitter._code.join "\n"
+      expect(code).toBe("""
+        import * as gregor1 from '../gregor1'
+        import * as keybase1 from 'github.com/keybase/client/go/protocol/keybase1'\n
+      """)
+
+      return
+
+    it "should ignore packages without an import_as field", () ->
+      imports = [
+        {
+          path: "../common.avdl",
+          type: "idl",
+        }
+      ]
+      emitter.emit_imports {imports, messages: {}, types: []}, 'location/of/my/output.go'
+      code = emitter._code.join "\n"
+      expect(code).toBe("")
+      return
+    return
+
+  describe "emit_typedef", () ->
+    it "should emit a type definition/alias", () ->
+      type = {
+        type: "record"
+        name: "BuildPaymentID"
+        fields: []
+        typedef: "string"
+      }
+      emitter.emit_typedef type
+      code = emitter._code.join "\n"
+
+      expect(code).toBe("""
+        export type BuildPaymentID = string\n
+      """)
+      return
+    return
+
+  describe "emit_record", () ->
+    it "should emit an object with primative value keys", () ->
+      record = {
+        type: "record"
+        name: "TestRecord"
+        fields: [
+          {
+            type: "string",
+            name: "statusDescription"
+          },
+          {
+            type: "boolean"
+            name: "isValidThing"
+          },
+          {
+            type: "long",
+            name: "longInt"
```

Comment:
Specifying `long` and using js's 53 bits is pretty sketch but I don't see a way around it. Ooh maybe `never`? As a way of discouraging users from accessing long (potentially corrupted) fields. Idk if that makes sense.

Reply:
We could make these [BigInts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt). That seems like a better long-term solution than `never`. I'm open to either though.

Reply:
That would require deserialization to know about types. If we could do that we could also deserialize `null` into `[]` when the type is a list.